### ISSUE TITLE
State dependant icons (motion, door / window sensors) replaced with mdi:bookmark

### ISSUE THREAD

**Issue by tomtom13** [benct/lovelace-multiple-entity-row#330]
Hi, after most recent update to HASS, the icons for stuff that has icon dependant of it's state (motion sensors / window & door sensors) stoped showing appropriate icons and shows mdi:bookmark. Saying that the state colour works. 

Current system info:
Core 2024.2.0
Supervisor 2024.01.1
Operating System 11.5
Frontend 20240207.0

**Comment by noharminthat** [comment]
I have the same issue with cover entities. The issue started occurring for me immediately upon upgrading to Core 2024.2.0.

**Comment by QRPp** [comment]
Same (or similar) here, started once updated to core 2024.2.0.

In more detail as per what I observe:
- The problem _doesn't apply_ to entities with icons overridden _either_ in their per-entity HA configuration or their per-entity `multiple-entity-row` configuration (then correctly shown).
- The problem _doesn't apply_ to _primary_ entities of any `multiple-entity-row`, affecting only their _listed_ entities, i.e., `entities: [{ entity: sensor.foo; icon: true; }]`.  Wrong icons are shown for the latter even when the same entity is also the same row's primary entity, i.e., the icon is correct at the start of the row but wrong later on the same row.
- The problem doesn't seem to depend on whether the icon of an entity is or is not state-dependent: e.g., all my temperature and humidity sensors listed via `multiple-entity-row` currently show up as `mdi:eye` at the moment, same as all my switches and binary sensors show up as `mdi:bookmark` there.
- The icon wrongly shown is the one of the corresponding entity's _domain_ (e.g., `mdi:eye` for [sensors](https://github.com/home-assistant/frontend/blob/20240207.1/src/common/const.ts#L133)) or `mdi:bookmark` for [domains with no icon](https://github.com/home-assistant/frontend/blob/20240207.1/src/common/const.ts#L91).  What appears somehow ignored is the icon that depends on the entity's _device class_.

**Comment by tankdeer** [comment]
Adding a little more context plus a visual to what others have observed.

![image](https://github.com/benct/lovelace-multiple-entity-row/assets/48227886/cea518de-db0d-4342-a516-06c52f5a1724)

Here you can see a mix and match of icons that show properly, and ones that don't. I hadn't fully figured out the pattern until the previous post. To add to what was mentioned before, with the exception of the window, all of the correctly showing icons in my case are template sensors. The icon changes based on state. The window is a normal Z-wave contact sensor set to show as window

**Comment by sulimoff** [comment]
Confirm, leakage  sensors also began to appear as bookmarks
![Screenshot_20240209-181010](https://github.com/benct/lovelace-multiple-entity-row/assets/56220294/05231e92-b85f-4e6f-874d-9a04c439fafe)

**Comment by aredon** [comment]
Very similar issue with battery status icons. They have been replaced with mdi:eye.

![image](https://github.com/benct/lovelace-multiple-entity-row/assets/1328757/12b3ef46-20bd-4a42-924f-5946eb372d17)

**Comment by bratanon** [comment]
A PR is submitted https://github.com/benct/lovelace-multiple-entity-row/pull/331

**Comment by tomtom13** [comment]
> Very similar issue with battery status icons. They have been replaced with mdi:eye.
> 
> ![image](https://private-user-images.githubusercontent.com/1328757/303772846-12b3ef46-20bd-4a42-924f-5946eb372d17.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc1MjA1ODgsIm5iZiI6MTcwNzUyMDI4OCwicGF0aCI6Ii8xMzI4NzU3LzMwMzc3Mjg0Ni0xMmIzZWY0Ni0yMGJkLTRhNDItOTI0Zi01OTQ2ZWIzNzJkMTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDlUMjMxMTI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzY3YmZjMmEyZjFmYmYzMGM5YTgwM2JmMmFjNzllNzVmY2YwNzUyYmNlZTFmN2Q3ZjVhYjMzM2FkMzJkN2EzMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LYHXl6V0YYvK34OANnOlFPnC75eVqNpEzij-HW76Q_Y)

TBH, this proves that somebody is always watching :D

**Comment by zhenyatk** [comment]
Some problem :(

**Comment by jeremysherriff** [comment]
Is there a way to test this while waiting for the PR to be merged?  I can't find the patched file (`index.js`) in my local filesystem

**Comment by ildar170975** [comment]
Tested it locally (made changes in an obfuscated js file by myself) with no success.
Do not think the added "`.hass=${this._hass}`" line helps, the changed `ha-state-icon` object seems to be needed here.

**Comment by jeremysherriff** [comment]
@ildar170975 I saw your comment on the PR. Where/how did you access the obfuscated file?

**Comment by ildar170975** [comment]
@jeremysherriff 
The obfuscated file is located in /config/www/community/lovelace-multiple-.... folder.
What I did was:
1. Remove gz-file.
2. Open js file in a text editor.
3. Find the place where this "`.hass=${this._hass}`" line should be added. Add the line (similarly like already present lines).
4. Save, clear cache, refresh a page.

Have to do this "brute" update since did not have the generated updated js-file.
But later after reading Dev blog started to doubt about a correctness of added change.

**Comment by QRPp** [comment]
> Tested it locally (made changes in an obfuscated js file by myself) with no success. Do not think the added "`.hass=${this._hass}`" line helps, the changed `ha-state-icon` object seems to be needed here. But ofc we should wait for a feedback from the author of the PR.

I've also tested the proposed fix locally by overriding the code directly using Chrome dev tools, and it surely fixes the problem as it should.  The `ha-state-icon` component is already in use, just not directly but wrapped via `state-badge`, so to speak.  Responded to the PR comment too.

How to:
- Load a HA panel that uses `multiple-entity-row` in Google Chrome, open dev tools.
- On the _Sources_ tab, search for `renderIcon` to find the snippet:
```
key: "renderIcon",
value: function(t, e) {
    return H(ye || (ye = we(['<state-badge\n [...]
```
- Start a local override, to which, in front of `.overrideIcon="', '"\n`, insert `.hass="', '"\n` and, further along the same line, change `t,` to `t, this._hass,`.  Be careful with all those commas and quote marks etc, the code editor wants to “help”, which here may be counter-productive.
- Save, reload and enjoy.

I hope this helps!

**Comment by ildar170975** [comment]
> * in front of `.overrideIcon="', '"\n`, insert `.hass="', '"\n` and, further along the same line, change `t,` to `t, this._hass,`.

Very tricky.
What I did was just adding this line:
![image](https://github.com/benct/lovelace-multiple-entity-row/assets/71872483/b2e586f9-394a-48ff-b6c4-60798d552300)
which seems to be wrong.
In fact, all these is a waste since we cannot get an updated js ))))

**Comment by jeremysherriff** [comment]
Order is important, as this is messing with concatenation of text strings within a function.
Adding a text version for easier copy/paste, and pointers to the location of the edits.
I actually edited the file in the HA filesystem, but forcing the browser to flush the original file from cache was harder than it should have been, so I recommend using the local override file as described above.

Relevant part of original line:
```
<state-badge\n            class="icon-small"\n            .stateObj="','"\n            .overrideIcon="','"\n            .stateColor="','"\n        ></state-badge>'])),t,!0===e.icon
```

Alter to:
```
<state-badge\n            class="icon-small"\n            .stateObj="','"\n            .hass="','"\n            .overrideIcon="','"\n            .stateColor="','"\n        ></state-badge>'])),t,this._hass,!0===e.icon
---------------------------------------------------------------------------------------^^^^^^^^^^^^^---------------------------------------------------------------------------------------------^^^^^^^^^^^
```

If you do use a local override file, you'll need to remember to remove that override once the update has been released.

**Comment by tomtom13** [comment]
I think I might be not the only person wondering,
BUT
assuming that this PR would be berged this very instant - how quickly we would be blessed with a release that includes it ?

**Comment by kineticscreen** [comment]
Not seeing any activity from @benct since December...

**Comment by tomtom13** [comment]
So ... is that repo dead ? Should we fork it ?

**Comment by benct** [comment]
I'll merge the PR and try to get a new version released as soon as possible.

**Comment by tomtom13** [comment]
heeeeeeyyyy ! lost person found ! thanks a lot bud !

**Comment by benct** [comment]
Should be fixed with version `4.5.1`.

**Comment by tomtom13** [comment]
FYI - works like a charm ! Thanks again.
